common/browser_process.go

```diff
@@ -179,41 +179,51 @@ func execute(
 	return command{cmd, done, stdout, stderr}, nil
 }
 
-// parseDevToolsURL grabs the websocket address from chrome's output and returns it.
-func parseDevToolsURL(ctx context.Context, cmd command) (wsURL string, _ error) {
-	type result struct {
-		devToolsURL string
-		err         error
-	}
-	c := make(chan result, 1)
+// parseDevToolsURL grabs the WebSocket address from Chrome's output and returns
+// it. If the process ends abruptly, it will return the first error from stderr.
+func parseDevToolsURL(ctx context.Context, cmd command) (string, error) { //nolint: cyclop
+	var (
+		lines = make(chan string)
+		errCh = make(chan error)
```

Comment:
The scanner would never return when we find the devToolsURL. To be on the safe side, let's buffer each one.

```suggestion
		lines = make(chan string, 1)
		errCh = make(chan error, 1)
```